android/src/main/java/io/customer/reactnative/sdk/NativeCustomerIOModule.kt

```diff
@@ -22,15 +22,13 @@ import io.customer.sdk.events.TrackMetric
 import io.customer.sdk.events.serializedName
 
 /**
- * Shared implementation logic for Customer.io Native SDK communication in React Native.
- * Contains actual business logic used by both old and new architecture [NativeCustomerIOModule] classes.
- * Handles SDK initialization, user identification, event tracking, and device management.
+ * React Native module implementation for Customer.io Native SDK
+ * using TurboModules with new architecture.
  */
-internal object NativeCustomerIOModuleImpl {
-    const val NAME = "NativeCustomerIO"
-
-    private val logger: Logger
-        get() = SDKComponent.logger
+class NativeCustomerIOModule(
```

Comment:
This is not entirely new file, but just merged version of new arch + common implementation of `NativeCustomerIOModule`